kes.user.js

```diff
@@ -427,6 +445,7 @@ function constructMenu(rawJSON, rawLayout, rawCSS) {
                     let label = document.createElement('p');
                     label.className = "kes-field-label";
                     label.innerText = json[it].fields[i].label;
+                    kesCheckSettingsOptionDisabled(label, json[it].fields[i].depends_value, getModSettings(json[it].namespace)[json[it].fields[i].depends_on], getModSettings(json[it].namespace).state);
```

Comment:
This seems unnecessarily verbose and I didn't get why it makes two calls to getModSettings when the contents of the object have already been set on line 439. We already have the entire contents of the settings for that namespace (ns), so no need to call the function again. Before we enter the switch statement, the contents of the settings object are parsed out with those rows of lets to make them available through the entirety of the switch statement. The only thing that changes is the field type, everything else remains the same. (Unless I'm misunderstanding)